cmd/cli/cmd/typeinstance/type_instance.go

```diff
@@ -57,16 +51,46 @@ func mapTypeInstanceToUpdateType(in *gqllocalapi.TypeInstance) gqllocalapi.Updat
 		return out
 	}
 
+	mapSpecValue := func() interface{} {
+		if in.LatestResourceVersion.Spec == nil {
+			return nil
+		}
+		return in.LatestResourceVersion.Spec.Value
+	}
+
+	mapBackendContext := func() interface{} {
+		if in.LatestResourceVersion.Spec.Backend == nil {
+			return nil
+		}
+		return in.LatestResourceVersion.Spec.Backend.Context
+	}
+
 	if in.LatestResourceVersion != nil {
 		out.TypeInstance = &gqllocalapi.UpdateTypeInstanceInput{
 			Attributes: mapAttrs(),
 			Value:      mapSpecValue(),
+			Backend: &gqllocalapi.UpdateTypeInstanceBackendInput{
```

Comment:
I would suggest moving the `&gqllocalapi.UpdateTypeInstanceBackendInput` to the `mapBackendContext` function, as otherwise you always have the `backend.context` in the output even if  the `in.LatestResourceVersion.Spec.Backend` was null.

Reply:
good point :+1: